src/main/java/frc/robot/subsystems/LedSubsystem.java

```diff
@@ -0,0 +1,124 @@
+package frc.robot.subsystems;
+
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+
+// our robot constants
+
+import frc.robot.Constants.LedConstants;
+
+
+
+public class LedSubsystem extends SubsystemBase {
+
+/**
+ * The LED subsystem provides all of the methods for driving the LEDs on the robot
+ */
+
+    private AddressableLED m_led;
+
+    private AddressableLEDBuffer m_ledBuffer;
+
+
+
+    public LedSubsystem() {
+
+        m_led = new AddressableLED(LedConstants.kLedPwmPort);
+
+        m_ledBuffer = new AddressableLEDBuffer(LedConstants.kNumOfLeds);
+        m_led.setLength(LedConstants.kNumOfLeds);
+
+        // setLedsMaroon();
+   
+        m_led.start();
+    }
+    
+
+
+    public int getNumOfLeds() {
+        return m_ledBuffer.getLength();
+    }
+
+
+
+    // all the various color and effect routines do not ned to do a .setData() call
+    //
+    // the periodic routine for the LedSubsystem will do it for you
+
+    public void setLedsOff() {
+
+        for (var i = 0 ; i < m_ledBuffer.getLength() ; i++) {
+   
+            m_ledBuffer.setRGB(i, 0, 0, 0);
+        }
+    }
+
+
+
+    public void setLedsWhite() {
+
+        for (var i = 0 ; i < m_ledBuffer.getLength() ; i++) {
+   
+            m_ledBuffer.setRGB(i, 180, 180, 180);
+        }
+    }
+
+
+
+    public void setLedsMaroon() {
+
+        for (var i = 0 ; i < m_ledBuffer.getLength() ; i++) {
+   
+            m_ledBuffer.setHSV(i, 2, 64, 22);       // determined with google colo picker at https://www.google.com/search?q=color+picker
+        }
+    }
+
+
+
+    public void setAllLedsRGB(int red, int green, int blue) {
+ 
+        for (var i = 0 ; i < m_ledBuffer.getLength() ; i++) {
+   
+            m_ledBuffer.setRGB(i, red, green, blue);
+        }
+    }
+
+
+
```

Comment:
Comment here explaining that this method sets a number of leds from a certain led, depending on count and index respectively.
```suggestion
.
```

Reply:
Good point, I can add that.  I'll add something for all of the less obvious routines.

If we think of other useful utility functions like this, we can add them into LedSubsystem as well.